crates/sui-network/src/state_sync/mod.rs

```diff
@@ -1034,10 +1034,17 @@ where
         debug_assert!(current.sequence_number().saturating_add(1) == next);
 
         // Verify the checkpoint
-        let checkpoint = {
+        let checkpoint = 'cp: {
             let checkpoint = maybe_checkpoint.ok_or_else(|| {
                 anyhow::anyhow!("no peers were able to help sync checkpoint {next}")
             })?;
+            // Skip verification for manually pinned checkpoints.
+            if pinned_checkpoints
+                .binary_search_by_key(checkpoint.sequence_number(), |(seq_num, _digest)| *seq_num)
```

Comment:
could `pinned_checkpoints` be a hash map? I guess not a big deal as the list won't be long

Reply:
Yeah I had the same thought and decided given this should have at most a few entries, vector would be more efficient